src/Mabiavalon.DockNC/Docking/Branch.cs

```diff
@@ -135,7 +151,7 @@ private void UpdateLogicalChildren(object oldItem, object newItem)
 
             if (oldChild != null)
             {
-                ((ISetLogicalParent) oldItem).SetParent(null);
+                //((ISetLogicalParent)oldItem).SetParent(null);
```

Comment:
Avalonia does not, unfortunately. The strange thing is that this exactly what `ContentControl` does in that order.